RIOT-OS/RIOT
# tests/periph/rtc: Add rtc mem survive reset test

### Contribution description
The `periph/rtc` test did not test if the rtc memory persisted across reboot. This PR aims to fix that.
The test is reworked to utilize the `shell` module. This is needed since testing across reboots means we have to clear RTC memory first, then write and verify, then reboot and verify again. In addition the test still tests the rtc alarms. 
Additionally this also changes the test to use `ztimer` instead of `xtimer`

Some parts of RIOT utilizes the backup registers in ways that will force a reset of the RTC and therefore of the registers. This test will fail on those since is clears and writes to all available registers. Ideas on how this should be handled is welcome!




### Testing procedure
When running the test application execute the following commands and verify the output

1. `rtc_clear` returns `RTC mem cleared`
2. `rtc_verify` returns `RTC mem content does not match`
3. `rtc_write` returns `RTC mem set`
4. `rtc_verify` returns `RTC mem OK`
5. `rtc_reboot` reboots device
6. `rtc_verify` returns `RTC mem OK`
7. `rtc_clear` returns `RTC mem cleared`. This is to clear RTC for the future.
8. `rtc_test_alarms` returns
```
This test will display 'Alarm!' every 2 seconds for 4 times
Clock value is now   2024-02-20 15:41:43.176
  Setting clock to   2020-02-28 23:59:57
Clock value is now   2020-02-28 23:59:57.176
  Setting alarm to   2020-02-28 23:59:59
   Alarm is set to   2020-02-28 23:59:59
  Alarm cleared at   2020-02-28 23:59:57.176
       No alarm at   2020-02-28 23:59:59.176
  Setting alarm to   2020-02-28 23:59:61

Alarm!
Alarm!
Alarm!
Alarm!
```

This procedure is also included in the `tests/01-run.py` so `make test` on a board that supports the `periph_rtc_mem` feature will execute the above steps.




### Issues/PRs references
